source/getting-started/user-guide/amds-interface.md

```diff
@@ -89,85 +89,84 @@ Place the code below into your custom user app init function. Modify the `GP3IO_
-
-If one sample delay is not acceptable for your application, keep the `AUTO_TX` mode disabled (i.e. set the define to `0`).
-You must manually request data at the start of the control task, and then wait for the data to arrive before running the controller.
-This reduces the time usable to compute the controller outputs since requesting and transmitting data from the AMDS is not instantaneous.
+Note that `get_data()` is **non-blocking** -- it will return the latest data that the AMDS sent, but will not trigger new data to be sent. This matches how the integrated ADCs work on the AMDC.
 
-To request new data from the AMDS, call `motherboard_request_new_data()` from `drv/motherboard`.
-Then, monitor the FPGA counters via `motherboard_get_counters()` and wait until the data has been received.
-Note that the counters increment for every byte of received data. See the digital interface data format [here](/accessories/amds/firmware/index).
+**NOTE**: The AMDS driver will make *all* data received from the mainboard available, even if the data became corrupted during transmission back to the AMDC.
 
-## Use Sampled Data
-
-After the data has been transmitted from the AMDS to the AMDC, it can be used by the user application.
-User code can read the raw 16-bit signed integer value as sampled on the AMDS sensor cards by using the following driver:
+Therefore, the user should also make use of the `check_data_validity()` function to verify that the data returned by `get_data()` was not corrupted during transmission. This function will return a byte (eight bits) where each bit represents the data validity for one of the eight channels on the AMDS. If this function returns 255, that implies that all channels returned valid data:
 
 ```C
-int motherboard_get_data(mb_channel_e channel, int32_t *out);
+uint8_t amds_check_data_validity(uint8_t port)
 ```
 
-Note that `get_data()` is **non-blocking** -- it will return the latest valid data that the AMDS sent, but will not trigger new data to be sent.
-This matches how the integrated ADCs work on the AMDC.
-
 For example, in a task callback:
 
 ```C
 void task_callback(void)
 {
     // ...
 
-    // Read in integer value sampled on the AMDS from channel 1:
     int32_t out;
-    int err = motherboard_get_data(MB_IN1, &out);
 
-    // Now, "out" contains the sign-extended 16-bit sampled value
+    // Check validity of latest data for the AMDS plugged into GPIO port 2
+    uint8_t valid = amds_check_data_validity(2);
+
+    if (valid & AMDS_CH_1_VALID_MASK != 0) {
+        // Yay! Channel 1's data is valid
+        // Read in integer value sampled on the AMDS (plugged into GPIO port 2) from channel 1:
+        int err = amds_get_data(2, AMDS_CH_1, &out);
+
+        // Now, "out" contains the sign-extended 16-bit sampled value
+    }
+    else {
+        // Else, you decide what to do if a given channel did not provide valid data this cycle
+
+        // ...
+    }
 
     // ...
 }
 ```
```

Comment:
Think hard about this code :) folks will copy-paste this code many times into their apps, so all the comments you have and exact format will become the exact code used by everyone.

For example, maybe do not use `2` constant, instead store as `const` variable..? etc

Reply:
Also, think about a user trying to read all 8 channels in their code... how would you recommend they write the c code? just copy paste this example 8 times...? ha, i hope not :)

Reply:
See... I am really hoping that people try to _understand_ the code instead of just copying and pasting 🥲

Reply:
But I am uncertain that that's a realistic wish